### PR TITLE
:recycle: Change board allocations from heap to stack

### PR DESCRIPTION
**Pull request by ickule**

### CODE REVIEW

**Review by kaoliiny — APPROVED** [review]

**Comment by kaoliiny** [comment]
If must also be resolved

**Comment by kaoliiny** [comment]
No. Please learn about static/dynamic memory allocation.

**Comment by kaoliiny** [comment]
Rule is simple: use the stack whenever you can

**Comment by ickule** [comment]
I believe that PR is made obsolete by https://github.com/ickule/yagolic/pull/31

File: include/copy_board.h [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #ifndef COPY_BOARD_H_INCLUDED
 #define COPY_BOARD_H_INCLUDED
 
-void copy_board(int rows, int cols, int *board, int *copy);
+void copy_board(int rows, int cols, int board[rows][cols], int copy[rows][cols]);
```
Comment (kaoliiny) [inline include/copy_board.h:4]: Does it even work?

Comment (kaoliiny) [reply include/copy_board.h:4]: Spoiler: no

File: include/spread_life.h [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #include "copy_board.h"
 #include "count_neighbours.h"
 
-void spread_life(int *board, int rows, int cols);
+void spread_life(int rows, int cols, int board[rows][cols]);
```
Comment (kaoliiny) [inline include/spread_life.h:9]: Please Google how to pass arrays into function

**Review by kaoliiny — APPROVED** [review]

**Review by kaoliiny — CHANGES_REQUESTED** [review]
fix comments

File: src/count_neighbours.c [hunk 3]
```diff
@@ -1,101 +1,101 @@
...
   int neighbours = 0;
   if (x == 0 && y == 0) {
     /* x = 0 and y = 0 so no (x - 1) nor (y - 1) */
     // clang-format off
     neighbours =
-      board[x * cols + (y + 1)] +
-      board[(x + 1) * cols + y] +
-      board[(x + 1) * cols + (y + 1)];
+      board[x][y+1] +
+      board[x+1][y] +
```
[the remark this answers is not in the document]
Comment (ickule) [reply src/count_neighbours.c:12]: Yeah, I figured out it is not really worth it keeping it a 2D matrix as the code got more complex.

File: include/copy_board.h (hunk 1 above)
Comment (ickule) [reply include/copy_board.h:4]: It does actually.
Point taken though, I did not intend to keep using this notation.

**Comment by ickule** [comment]
I fixed all the comments and rebased with the latest additions.

Is that what you had in mind?

I did not find a way to not have a fixed size for the board and I could not have the function directly initialised the cells array since their scope are limited to the current function when allocated from stack as far as I understand. 
I am happy to take in suggestions if there is something I missed on how to do that.

As I mentioned in https://github.com/ickule/yagolic/pull/31 I do not like the idea of a fixed board like this and the previous version allocated from heap feels simpler and clearer.

**Pull request closed without merging**